src/frontend/src/optimizer/plan_node/stream_table_scan.rs

```diff
@@ -42,7 +42,7 @@ impl StreamTableScan {
             ctx,
             logical.schema().clone(),
             logical.base.pk_indices.clone(),
-            Distribution::Single,
+            Distribution::HashShard(logical.base.pk_indices.clone()),
```

Comment:
Could you please also change the distribution in the Java frontend? I'm afraid the current e2e result cannot cover the case.

Reply:
I've added a workaround for Java frontend in fragmenter (because I'm not familiar with Java part 😅). Will remove this after we deprecate Java frontend.